app/components/spotlight/analytics/dashboard_component.rb

```diff
@@ -9,7 +9,63 @@ class DashboardComponent < ViewComponent::Base
       def initialize(current_exhibit:)
         super
         @current_exhibit = current_exhibit
-        @dates = { 'start_date' => '365daysAgo', 'end_date' => 'today' }
+        @default_start_date = 1.year.ago.to_date.to_s
+        @default_end_date = Time.zone.today.to_date.to_s
+      end
+
+      def before_render
+        flash[:error] = nil
+        @dates = { 'start_date' => @default_start_date,
+                   'end_date' => @default_end_date }
+
+        validate_dates
+      end
+
+      def validate_dates
+        if params[:start_date] && params[:end_date] && params[:start_date] > params[:end_date]
+          flash[:error] = I18n.t('spotlight.dashboards.analytics.error_heading', date: "#{params[:start_date]} > #{params[:end_date]}")
+          return
+        end
+        update_dates
+      end
+
+      def update_dates
+        date('start_date') if params[:start_date] && params[:start_date] >= min_date
+        date('end_date') if params[:end_date] && params[:end_date] <= max_date
```

Comment:
Should the checks happening here -- that the start and end dates conform to the configured min/max be part of the validation step?